simulation/results/report.md

```diff
@@ -3,42 +3,42 @@
 <!-- Run `cargo test -p simulation --release` to regenerate this report. -->
 
 ```
-                   all_nodes_500[CONCURRENCY_LIMITER_PIN_UNTIL_ERROR]:	success=50%	client_mean=600ms	server_cpu=1200s	client_received=2000/2000	server_resps=2000	codes={200=1000, 500=1000}
-                       all_nodes_500[CONCURRENCY_LIMITER_ROUND_ROBIN]:	success=50%	client_mean=600ms	server_cpu=1200s	client_received=2000/2000	server_resps=2000	codes={200=1000, 500=1000}
+                   all_nodes_500[CONCURRENCY_LIMITER_PIN_UNTIL_ERROR]:	success=74.2%	client_mean=6.093870532s	server_cpu=1200s	client_received=2000/2000	server_resps=2000	codes={200=1483, 500=517}
```

Comment:
Woah, the client mean increased from 0.60s to 6.09s. That looks concerning.

Reply:
It looks like the number of failures have been halved. I'd take that trade.

Reply:
That is 50% slower than Dialogue, which is a bit strange since I think our behavior should in theory be pretty much identical for PIN_UNTIL_ERROR?

Reply:
I'm not sure how to interpret the graphs. I guess we want to add logging to track the permit count and total permit changes to better understand what's taking so long.

![all_nodes_500 CONCURRENCY_LIMITER_PIN_UNTIL_ERROR](https://user-images.githubusercontent.com/766615/102659387-2af60780-412e-11eb-8b9e-53a202abca97.png)

Reply:
I think the difference here is just that Dialogue and c-r-r pinned to different collections of nodes. While we can expect the balanced selectors to end up behaving roughly the same in the limit, the single choice of a node to pin on in each fo the 10 clients with pin-until-error ends up adding a bunch of variance.

Reply:
And the slowdown from 600ms to 6s is just the client slowing down requests made early while the server was failing, which results in the success rate jumping from 50% to 75%.

Reply:
OK, this sounds reasonable. Let's merge and see how it looks in practice.